File: front/src/app/pages/client/chatbot/services/chatbot.service.ts
import { Injectable } from '@angular/core';
import { HttpClient } from '@angular/common/http';
import { Observable } from 'rxjs';
import { ChatbotRequest, ChatbotResponse, ChatbotSession, ChatbotMessage, ChatbotSessionTitle } from '../models/chatbot.model';

@Injectable({ providedIn: 'root' })
export class ChatbotService {
  private readonly BASE = '/api/v1/chatbot';

  constructor(private http: HttpClient) {}

  sendMessage(request: ChatbotRequest): Observable<ChatbotResponse> {
    return this.http.post<ChatbotResponse>(`${this.BASE}/message`, request);
  }

  getSessions(status?: string, page?: number, size?: number): Observable<ChatbotSession[]> {
    let params: any = {};
    if (status) params.status = status;
    if (page !== undefined) params.page = page;
    if (size !== undefined) params.size = size;
    return this.http.get<ChatbotSession[]>(`${this.BASE}/sessions`, { params });
  }

  createSession(): Observable<ChatbotSession> {
    return this.http.post<ChatbotSession>(`${this.BASE}/sessions`, {});
  }

  getSessionMessages(sessionId: number, page?: number, size?: number): Observable<any> {
    let params: any = {};
    if (page !== undefined) params.page = page;
    if (size !== undefined) params.size = size;
    return this.http.get<any>(`${this.BASE}/sessions/${sessionId}/messages`, { params });
  }

  updateTitle(sessionId: number, title: string): Observable<ChatbotSession> {
    return this.http.patch<ChatbotSession>(`${this.BASE}/sessions/${sessionId}/title`, { title } as ChatbotSessionTitle);
  }

  deleteSession(sessionId: number): Observable<void> {
    return this.http.delete<void>(`${this.BASE}/sessions/${sessionId}`);
  }

  archiveSession(sessionId: number): Observable<ChatbotSession> {
    return this.http.patch<ChatbotSession>(`${this.BASE}/sessions/${sessionId}/archive`, {});
  }

  restoreSession(sessionId: number): Observable<ChatbotSession> {
    return this.http.patch<ChatbotSession>(`${this.BASE}/sessions/${sessionId}/restore`, {});
  }

  escalateSession(sessionId: number): Observable<ChatbotSession> {
    return this.http.post<ChatbotSession>(`${this.BASE}/sessions/${sessionId}/escalate`, {});
  }
}

File: back/src/main/java/com/okane/service/chatbot/ChatbotService.java
package com.okane.service.chatbot;

import com.okane.dto.chatbot.ChatbotRequestDTO;
import com.okane.dto.chatbot.ChatbotResponseDTO;
import com.okane.dto.chatbot.ChatbotSessionDTO;
import com.okane.dto.chatbot.ChatbotMessageDTO;
import org.springframework.data.domain.Page;

import java.util.List;

public interface ChatbotService {
    ChatbotResponseDTO processMessage(ChatbotRequestDTO request, Long clientId);
    Page<ChatbotMessageDTO> getSessionMessages(Long sessionId, Long clientId, int page, int size);
    List<ChatbotSessionDTO> getSessions(Long clientId, String statusFilter);
    ChatbotSessionDTO createSession(Long clientId);
    ChatbotSessionDTO updateTitle(Long sessionId, Long clientId, String title);
    void deleteSession(Long sessionId, Long clientId);
    ChatbotSessionDTO archiveSession(Long sessionId, Long clientId);
    ChatbotSessionDTO restoreSession(Long sessionId, Long clientId);
    ChatbotSessionDTO escalateSession(Long sessionId, Long clientId);
}
